### PR TITLE
[#385] Adds a user preference option to remove fixed positioned elements during full page screenshots

### PR DESCRIPTION
**Pull request by crperezt**
## Description:
- Previously, full page screenshots would hide `position: fixed` elements to avoid having the elements repeated. Now users can choose whether to do this on know through the user preferences. Resolves issue #385.

## Changes I Made:
- Added checkbox in user preferences for removing fixed positioned elements in full page screenshots.
- Added state for the new user preference to browser reducer.
- Added logic to WebView component and screenshotUtil.js to only remove fixed positioned elements when option is checked.

### CODE REVIEW

File: desktop-app/app/components/UserPreferences/index.js [hunk 1]
```diff
@@ -86,6 +86,25 @@ export default function UserPreference({
...
         </div>
+        <div>
+          <FormControlLabel
+            control={
+              <Checkbox
+                checked={userPreferences.removeFixedPositionedElements || false}
+                onChange={e =>
+                  onChange('removeFixedPositionedElements', e.target.checked)
+                }
+                name="Remove fixed position elements from full page screenshot"
```
[the remark this answers is not in the document]
Comment (manojVivek) [reply desktop-app/app/components/UserPreferences/index.js:97]: Yes agreed, it should be enabled by default unless explicitly disabled.

[the remark this answers is not in the document]
Comment (crperezt) [reply desktop-app/app/components/UserPreferences/index.js:97]: Gotcha, I missed the fact that quick screenshot also removes the fixed positioned elements. Will edit, and make hiding the default.

File: desktop-app/app/components/WebView/screenshotUtil.js [hunk 2]
```diff
@@ -135,14 +141,16 @@ class WebViewUtils {
...
 
-    await this.webView
-      .executeJavaScript(
-        `
-          document.body.classList.add('responsivelyApp__ScreenshotInProgress');
-          responsivelyApp.hideFixedPositionElementsForScreenshot();
-        `
-      )
-      .catch(captureOnSentry);
+    if (removeFixedPositionedElements) {
```
[the remark this answers is not in the document]
Comment (crperezt) [reply desktop-app/app/components/WebView/screenshotUtil.js:147]: Ok, just fixed this, and changed the scrollbar portion to do: 
`.body::-webkit-scrollbar {
    display: none;
 }`
as you suggested.

File: desktop-app/app/components/UserPreferences/index.js [hunk 3]
```diff
@@ -86,6 +86,25 @@ export default function UserPreference({
...
+                color="primary"
+              />
+            }
+            label={
+              <span className={cx(styles.preferenceName)}>
+                Remove fixed position elements from full page screenshot
```
[the remark this answers is not in the document]
Comment (crperezt) [reply desktop-app/app/components/UserPreferences/index.js]: Done.

File: desktop-app/app/components/UserPreferences/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (crperezt) [reply desktop-app/app/components/UserPreferences/index.js]: Done.

File: desktop-app/app/components/WebView/screenshotUtil.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (manojVivek) [reply desktop-app/app/components/WebView/screenshotUtil.js:147]: Guys, if we just add a CSS to the `body` selector, how do we restore the original style after the screenshot is taken? 

The scrollbar will stay hidden even after the screen is taken. No? 

I would suggest, sticking to the classname-based approach. Let me your thoughts.

[the remark this answers is not in the document]
Comment (jzabala) [reply desktop-app/app/components/WebView/screenshotUtil.js:147]: @manojVivek 

> The scrollbar will stay hidden even after the screen is taken. No?

No, the key of the added css

https://github.com/responsively-org/responsively-app/blob/c1ee0b084435f308cfa3ab3b4871b7629e83fc31/desktop-app/app/components/WebView/screenshotUtil.js#L137

is used later to remove it

https://github.com/responsively-org/responsively-app/blob/c1ee0b084435f308cfa3ab3b4871b7629e83fc31/desktop-app/app/components/WebView/screenshotUtil.js#L167

[the remark this answers is not in the document]
Comment (manojVivek) [reply desktop-app/app/components/WebView/screenshotUtil.js:147]: Got it! My bad, missed that one.

File: desktop-app/app/components/WebView/screenshotUtil.js [hunk 4]
```diff
@@ -124,9 +131,11 @@ class WebViewUtils {
     await _delay(200);
   }
 
-  async hideScrollbarAndFixedPositionedElements(): Promise<string> {
+  async hideScrollbarAndFixedPositionedElements(
+    removeFixedPositionedElements: boolean
+  ): Promise<string> {
     const key = await this.webView.insertCSS(`
-      .responsivelyApp__ScreenshotInProgress::-webkit-scrollbar {
+      .body::-webkit-scrollbar {
```
Comment (jzabala) [inline desktop-app/app/components/WebView/screenshotUtil.js:138]: Here should be `body` alone. Without the `.`.

Comment (crperezt) [reply desktop-app/app/components/WebView/screenshotUtil.js]: Good catch, fixed.

**Comment by crperezt** [comment]
Thanks for letting me contribute! 👍

**Comment by manojVivek** [comment]
@crperezt Thank you so much for the effort you have put into this, merging this now. 🎉 

Looking forward to more like this in the future. 🙌

**Comment by manojVivek** [comment]
@all-contributors Please add @crperezt for code.

**Comment by manojVivek** [comment]
@all-contributors Please add @crperezt for code.

**Comment by manojVivek** [comment]
@crperezt Two things:

1. Can you let us know your Twitter handle? We would love to give a shout out for your work from our twitter account.
2. Please join our community on Gitter - https://gitter.im/responsively-app/, for any discussions in the future.

**Comment by crperezt** [comment]
Awesome! Very much appreciated. Twitter handle is: @carlosPerezDev.

**Pull request merged**